### PR TITLE
Reformat H3 index config spec

### PR DESCRIPTION
**Pull request by yupeng9**
Renamed the fields of H3 index config for style matching
Addressed comments in https://github.com/apache/incubator-pinot/pull/6397/files

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java [hunk 1]
```diff
@@ -21,30 +21,48 @@
...
 import com.google.common.base.Preconditions;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.pinot.spi.config.BaseJsonConfig;
 
 
-public class H3IndexConfig extends BaseJsonConfig {
+public class H3IndexColumn extends BaseJsonConfig {
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java:31]: Do we still need this class as a config class (json serializable), or this is just a helper class now? If the latter, maybe move this class to core for clarity?

Comment (yupeng9) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java:31]: I think we can keep it as it's possible to introduce a POJO for index column and use polymorphism to support this H3indexColumn

Comment (kishoreg) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java:31]: +1 on keeping this as a BaseJsonConfig, let's make sure we follow the same convention. should this be H3IndexConfig or H3IndexColumn or just H3Index?

File: pinot-tools/src/main/resources/examples/batch/starbucksStores/starbucksStores_offline_table_config.json [hunk 2]
```diff
@@ -11,16 +11,18 @@
   },
   "tenants": {
   },
+  "fieldConfigList": [
+    {
+      "name": "location_st_point",
+      "encodingType":"RAW",
+      "indexType":"H3",
+      "properties": {
```
Comment (kishoreg) [inline pinot-tools/src/main/resources/examples/batch/starbucksStores/starbucksStores_offline_table_config.json:19]: this does not match the H3ColumnIndex as BaseConfig, right?

**Review by kishoreg — APPROVED** [review]
code looks good. we need to figure out the config convention.

File: pinot-tools/src/main/resources/examples/batch/starbucksStores/starbucksStores_offline_table_config.json (hunk 2 above)
Comment (yupeng9) [reply pinot-tools/src/main/resources/examples/batch/starbucksStores/starbucksStores_offline_table_config.json:19]: It uses the properties to construct.

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java (hunk 1 above)
Comment (yupeng9) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java:31]: I feel `H3IndexColumn ` is better. In fact, I think it's possible to make it a subclass of `FieldConfig` by adding `@Type` per [this](https://stackoverflow.com/questions/30362446/deserialize-json-with-jackson-into-polymorphic-types-a-complete-example-is-giv).

**Pull request merged**